### PR TITLE
Dev 1583 sign up

### PR DESCRIPTION
**Pull request by sierra-henderson**
[changelog]
added: DEV-1583 Partner can signup/register for the partner portal

### CODE REVIEW

**Comment by sierra-henderson** [comment]
This PR contains the sign up screen, along with a temporary message on the bottom noting that a verification has been sent to a user's email. The verification modal and sign in routing will be addressed in future tickets.

File: src/app/graphql/cache.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { InMemoryCache } from '@apollo/client/core'
+
+const cache = new InMemoryCache({
+  typePolicies: {
```
Comment (jazzdragon) [inline src/app/graphql/cache.ts:5]: Please remove these typePolicies, they're not relevant to the partner portal.

File: src/app/graphql/mutations.graphql [hunk 2]
```diff
@@ -0,0 +1,188 @@
+mutation uploadUrl($input: UploadUrlArgsInput!) {
```
Comment (jazzdragon) [inline src/app/graphql/mutations.graphql:1]: We don't need all of these mutations, please remove.

File: src/app/shared/interfaces/maintenance.interface.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export interface MaintenanceMode {
```
Comment (jazzdragon) [inline src/app/shared/interfaces/maintenance.interface.ts:1]: Why did you copy this over?

File: src/app/shared/interfaces/user-role.interface.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export enum UserRole {
```
Comment (jazzdragon) [inline src/app/shared/interfaces/user-role.interface.ts:1]: Why did you copy this over?

File: src/app/shared/interfaces/user-session.interface.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    [sfOrgAttr]: string;
+  };
+}
+
+export interface Callback {
+  onSuccess: (success?: string) => void;
+  onFailure: (error: any) => void;
+}
+
+export interface RememberedDevice {
```
Comment (jazzdragon) [inline src/app/shared/interfaces/user-session.interface.ts:60]: Not needed

File: src/app/shared/interfaces/user-session.interface.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  Device: {
+    DeviceAttributes: DeviceAttributes[];
+    DeviceCreateDate: number;
+    DeviceKey: string;
+    DeviceLastAuthenticatedDate: number;
+    DeviceLastModifiedDate: number;
+  };
+}
+
+export interface DeviceAttributes {
```
Comment (jazzdragon) [inline src/app/shared/interfaces/user-session.interface.ts:70]: Not needed

File: src/app/shared/interfaces/user-session.interface.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
+import { CognitoUser } from '@aws-amplify/auth'
+import { UserRole } from './user-role.interface'
+
+export const sfOrgAttr = 'custom:sf_org'
+export const contactIdAttr = 'custom:contact_id'
+
+// CognitoUser definition is missing "challengeName"
+// https://github.com/aws-amplify/amplify-js/issues/3316
+type UserChallenge =
```
Comment (jazzdragon) [inline src/app/shared/interfaces/user-session.interface.ts:9]: Is this needed?

File: src/app/shared/services/app.service.ts [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+export enum ScreenSizeBreakpoints {
+  XXL = '2xl',
+  XL = 'xl',
+  LG = 'lg',
+  MD = 'md',
+  SM = 'sm',
+  MOBILE = 'mobile',
+}
+
+export interface IConfigSettings {
```
Comment (jazzdragon) [inline src/app/shared/services/app.service.ts:17]: Some of these are not relevant

File: src/app/shared/services/maintenance.service.ts [hunk 9]
```diff
@@ -0,0 +1,102 @@
+import { HttpClient, HttpHeaders } from '@angular/common/http'
```
Comment (jazzdragon) [inline src/app/shared/services/maintenance.service.ts:1]: why did you copy this over?

File: src/app/shared/services/routing-data.service.ts [hunk 10]
```diff
@@ -0,0 +1,51 @@
+import { Injectable } from '@angular/core'
```
Comment (jazzdragon) [inline src/app/shared/services/routing-data.service.ts:1]: why did you copy this over?

File: src/app/shared/services/user-role.service.ts [hunk 11]
```diff
@@ -0,0 +1,29 @@
+import { LocalStorageManagementService } from './local-storage-management.service'
```
Comment (jazzdragon) [inline src/app/shared/services/user-role.service.ts:1]: why did you copy this?

File: src/app/shared/services/user-session.service.ts [hunk 12]
```diff
@@ -0,0 +1,248 @@
+import { AppService } from './app.service'
```
Comment (jazzdragon) [inline src/app/shared/services/user-session.service.ts:1]: Please go over this and remove whatever is not relevant

File: src/assets/app.config.json [hunk 13]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (jazzdragon) [inline src/assets/app.config.json:1]: we don't need all of this.

File: src/environments/environment.ts [hunk 14]
```diff
@@ -1,10 +1,15 @@
...
 // The list of file replacements can be found in `angular.json`.
 
 export const environment = {
-  production: false
-};
+  production: false,
+  stage: false,
+  dev: false,
+  local: true,
+  amplitudeApiKey: '93cbe14f0b6783fa3a4c6f9223875a10',
```
Comment (jazzdragon) [inline src/environments/environment.ts:11]: Not relevant

File: src/main.ts [hunk 15]
```diff
@@ -1,12 +1,57 @@
...
+let env: string
+if (environment.production) {
+  env = 'production'
+} else if (environment.stage) {
+  env = 'stage'
+} else {
+  env = 'dev'
+}
+
+Sentry.init({
```
Comment (jazzdragon) [inline src/main.ts:17]: Remove until we add sentry to the repo

File: src/app/app.component.ts [hunk 16]
```diff
@@ -1,10 +1,79 @@
...
+    private appService: AppService,
+    private apollo: Apollo,
+    private maintenanceService: MaintenanceService,
+  ) {
+    this.toasterConfig = this.feedback.config
+    this.appService.screenWidth = window.innerWidth
+    this.appService.screenHeight = window.innerHeight
+
+    this.routingService.subscribeToTitle()
+    this.maintenanceService.fetchMaintenanceMode()
```
Comment (jazzdragon) [inline src/app/app.component.ts:35]: remove

**Review by jazzdragon — CHANGES_REQUESTED** [review]
- Please make sure we don't add noise by copying over code that will be dead because it's not relevant to this repo.
- Fix tests

File: src/app/shared/interfaces/maintenance.interface.ts (hunk 3 above)
Comment (sierra-henderson) [reply src/app/shared/interfaces/maintenance.interface.ts:1]: Wasn't sure if we'd be using maitenance mode in this app, I'll remove

File: src/app/shared/interfaces/user-session.interface.ts (hunk 7 above)
Comment (sierra-henderson) [reply src/app/shared/interfaces/user-session.interface.ts:9]: It is used in type `ICognitoUser` which is used in the `UserSessionService`

**Review by jazzdragon — APPROVED** [review]
Looks better. A couple of things:

- The header font seems to not load: https://capture.dropbox.com/WsQubaIIuls8EDiK?src=ss
- Please add the Uqual logo and fix the title (https://capture.dropbox.com/ckJdealvfCyEoiox?src=ss)

**Pull request merged**